packages/republik-crowdfundings/lib/scheduler/payments/sftp.ts

```diff
@@ -0,0 +1,92 @@
+import path from 'path'
+import { PassThrough } from 'stream'
+import Client from 'ssh2-sftp-client'
+import toArray from 'stream-to-array'
+
+declare module 'tar' {
+  class Parse {}
+}
+
+export async function getConnectedSftpClient(
+  c: Client.ConnectOptions,
+): Promise<Client> {
+  const sftp = new Client()
+
+  await sftp.connect(c)
+
+  return sftp
+}
+
+export async function listSftpFiles(
+  client: Client,
+  basePath = './',
+): Promise<string[]> {
+  const directories = await client.list(basePath)
+
+  const dirContents = (
+    await Promise.all(
+      directories
+        .filter((file) => file.type === 'd')
+        .map((f) => {
+          return f
+        })
+        .filter((f) => f.name.match(/^[^.]/))
+        .map(async (dir) => {
+          const dirPath = path.join(basePath, dir.name)
+          return listSftpFiles(client, dirPath)
+        }),
+    )
+  ).flat()
+
+  const camtFiles = await client
+    .list(basePath, 'camt.053*')
+    .then((fileInfos) => {
+      return fileInfos.map((fileInfo) => {
+        return path.join(basePath, fileInfo.name)
+      })
+    })
+  return [...camtFiles, ...dirContents]
+}
+
+export interface SftpFile {
+  fileName: string
+  buffer: Buffer
+}
+
+export function loadSftpFiles({
+  fileNames,
+  client,
+}: {
+  fileNames: string[]
+  client: Client
+}): Promise<SftpFile[]> {
+  return fileNames.reduce(
+    async (promisePreviousResult: Promise<SftpFile[]>, fileName) => {
+      const previousResult = await promisePreviousResult
+      const file = await request(client, fileName)
+
+      return previousResult.concat(file)
+    },
+    Promise.resolve([]),
+  )
+}
+
+async function request(client: Client, fileName: string): Promise<SftpFile> {
+  const stream = new PassThrough()
+  client.get(fileName, stream)
+  const buffer = await streamToBuffer(stream)
```

Comment:
have you tried:
```suggestion
  const buffer = await client.get(fileName)
```
Docs says that you can omit dst / passing a stream and directly get a buffer:
> get(path, dst, options) ==> String|Stream|Buffer
Retrieve a file from a remote SFTP server. The dst argument defines the destination and can be either a string, a stream object or undefined. If it is a string, it is interpreted as the path to a location on the local file system (path should include the file name). If it is a stream object, the remote data is passed to it via a call to pipe(). If dst is undefined, the method will put the data into a buffer and return that buffer when the Promise is resolved. If dst is defined, it is returned when the Promise is resolved.

Reply:
ARE YOU SERIOUS?!?